File: src/main/java/io/github/sd3v/mflashcardsbe/config/AppConfig.java
package io.github.sd3v.mflashcardsbe.config;

import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoClients;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.config.CorsRegistry;
import org.springframework.web.reactive.config.WebFluxConfigurer;

@Configuration
public class AppConfig {

  @Bean
  public MongoClient mongoClient() {
    return MongoClients.create();
  }

  @Bean
  public WebFluxConfigurer webFluxConfigurer() {
    return new WebFluxConfigurer() {
      @Override
      public void addCorsMappings(CorsRegistry registry) {
        registry.addMapping("/**").allowedOrigins("http://localhost:7700");
      }
    };
  }
}

File: src/main/kotlin/io/github/sd3v/mflashcardsbe/config/AppConfig.kt
package io.github.sd3v.mflashcardsbe.config

import com.mongodb.reactivestreams.client.MongoClient
import com.mongodb.reactivestreams.client.MongoClients
import org.springframework.boot.autoconfigure.web.reactive.WebFluxAutoConfiguration.WebFluxConfig
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.config.CorsRegistry
import org.springframework.web.reactive.config.WebFluxConfigurer

@Configuration
class AppConfig {

    @Bean
    fun mongoClient(): MongoClient = MongoClients.create()

    @Bean
    fun corsConfigurer(): WebFluxConfigurer =
        object : WebFluxConfigurer {
                override fun addCorsMappings(registry: CorsRegistry) {
                    registry.addMapping("/api/**").allowedOrigins("http://localhost:7700")
                }
        }
}
